packages/mongodb-dialects/mongosh/src/main/kotlin/com/mongodb/jbplugin/dialects/mongosh/backend/MongoshBackend.kt

```diff
@@ -0,0 +1,187 @@
+package com.mongodb.jbplugin.dialects.mongosh.backend
+
+import com.mongodb.jbplugin.mql.*
+import org.bson.types.ObjectId
+import org.owasp.encoder.Encode
+import java.math.BigDecimal
+import java.math.BigInteger
+import java.time.Instant
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.*
+
+private const val MONGODB_FIRST_RELEASE = "2009-02-11T18:00:00.000Z"
+
+/**
+ * @param context
+ */
+class MongoshBackend(private val context: Context = DefaultContext()) : Context by context {
```

Comment:
Context by context means that MongoshBackend implements Context, and delegates all the methods of Context to the variable context. It's done like this for convenience because we need to register variables and constants into the backend from within the mongosh formatter